src/components/cluster/Environment.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { Component } from 'react'
+import { showError, Pencil, useForm, Progressing, CustomPassword, VisibleModal, sortCallback } from '../common';
+import { List, CustomInput } from '../globalConfigurations/GlobalConfiguration'
+import { getClusterList, saveCluster, updateCluster, saveEnvironment, updateEnvironment, getEnvironmentList, getCluster, retryClusterInstall } from './cluster.service';
+import { ReactComponent as Close } from '../../assets/icons/ic-close.svg';
+import { EnvironmentValue,EnvironmentProps, EnvironmentState } from './cluster.type'
+import { toast } from 'react-toastify';
+
+const DefaultEnvironmentValue = {
+    environment_name: "",
+    namespace: "",
+    isProduction: "true",
+    prometheus_endpoint: "",
+    cluster_id: undefined
+}
+
+export class Environment extends Component<EnvironmentProps, EnvironmentState> {
+    constructor(props) {
+        super(props)
+        this.state = {
+            environment: [],
+            id: undefined,
+            loading: false,
+            isclosed: false,
+            saveLoading: false,
+            isError: {
+                environment_name: "",
+                namespace: "",
+            },
+            form: {
+                ...DefaultEnvironmentValue
+            },
+        }
+        this.handleInputChange = this.handleInputChange.bind(this)
+        this.handleOnSubmit = this.handleOnSubmit.bind(this);
+    }
+
+    componentDidMount() {
+        this.fetchEnvironmentList();
+    }
+
+    fetchEnvironmentList() {
+        getEnvironmentList().then((response) => {
+            let environment = response.result?.find(item => item.active);
+            this.setState({
+                environment: environment,
+            })
+            if(!environment){
+                environment = {
+                    ...DefaultEnvironmentValue
+                }
+            }
+            let isError = this.getFormErrors(false, environment)
+            this.setState({
+                environment: response.result || "",
+                loading: false,
+                saveLoading: false,
+                form: environment,
+                isError: isError,
+                id: this.state.id
```

Comment:
what is id here?
initially value is undefined.
here setting state with same `undefined` value